### ISSUE TITLE
function 'concat_ws' should ignore NULL in last argument

### ISSUE THREAD

**Issue by mateuszkj** [apache/arrow-datafusion#2100]
**Describe the bug**
Function *concat_ws* is not ignoring NULL in last argument and seperator text is added at the end of the result.

**To Reproduce**
```sql
SELECT concat_ws('|', 'a', NULL);
-- Result: a|
```

Tested on master (refs 73ea6e1)

**Expected behavior**
```sql
SELECT concat_ws('|', 'a', NULL);
-- Result: a
```
There should not be spectator added as the end of result.

**Additional context**
There is test case for ignoring NULLs in arguments but only in the middle: https://github.com/apache/arrow-datafusion/blob/master/datafusion/tests/sql/expr.rs#L553=

Function description says NULLs should be ignored: https://github.com/apache/arrow-datafusion/blob/master/datafusion-physical-expr/src/string_expressions.rs#L326=

**Comment by doki23** [comment]
I think we can use [join](https://doc.rust-lang.org/std/primitive.slice.html#method.join) to replace `String::push_str`.

**Comment by WinkerDu** [comment]
A raw fix here #2128
